internal/provider/resource_apply_module.go

```diff
@@ -585,3 +833,55 @@ func (t *applyModuleResource) copyRunVariablesToInput(ctx context.Context, list
 
 	return result, nil
 }
+
+// toProviderOutputVariables converts SDK variables from a finished run to the types the provider can return to Terraform.
+func (t *applyModuleResource) toProviderOutputVariables(
+	ctx context.Context,
+	arg []sdktypes.RunVariable,
+) (basetypes.ListValue, diag.Diagnostics) {
+	variables := []types.Object{}
+
+	for _, variable := range arg {
+		val := ""
```

Comment:
Should the value be made optional in the schema above so the value could be nil same perhaps for namespace path?

Reply:
This function is used for the resolved_variables (aka ResolvedVariables) attribute, which is computed.  Isn't optional only for input attributes?  I doubt Terraform would allow optional on a field of an element in a list-type computed output attribute.  Brandon has not mentioned any difficulty with this in all the review rounds so far.  A change would require splitting the RunVariableModel type, one for input and one for output, but that's not difficult.  Brandon, your guidance, please?

Reply:
If you mark something as `optional` in the schema, it means the user can pass a value in to have field.

Reply:
Ah, let's leave it as-is then.